test/unit/test_smdataparallel.py

```diff
@@ -104,7 +104,7 @@ def test_smdataparallel_run_multi_node_python(
                 "-x",
                 smdataparallel_flag,
                 "-x",
-                "FI_PROVIDER=sockets",
+                "FI_PROVIDER=efa",
```

Comment:
Do we need some kind of test to check if "efa" is getting picked up, will depend on instance we pick? but would be nice to have.

Reply:
by "efa" getting picked, do you mean searching for the logs after the SM Training job is started to make sure EFA is being used? or some other way of testing it?

Reply:
In this Training toolkit repo, we dont have a integration test for smddp. As in, SM training job isn't specifically invoked. As you can see in this PR code change, we have a unit test taht checks whether the parameters of the training job are correct or not. But actual SM training isn't invoked. We rely on SM SDK repo integration tests for the same. But it is a nice suggestion to check for EFA here and/or in SM Py SDK repo.

Reply:
Ok sounds good.